### PR TITLE
T1339

### PR DESCRIPTION
**Pull request by k88hudson**
Draggable hooks

### CODE REVIEW

File: templates/supported/newscaster/butter/template.js [hunk 1]
```diff
@@ -23,10 +24,114 @@ document.addEventListener( "DOMContentLoaded", function( e ){
           }
         });
 
+        // EDITING HOOKS
+        // Would be good if these were part of a library in shared, instead of existing here.
+        // Should be updated to a hook syntax like hook("image2", myUpdateFunction) or something
+        // Add listeners for future track events.
+          butter.listen("trackeventadded", updateFunction);
```
Comment (secretrobotron) [inline templates/supported/newscaster/butter/template.js]: Can you fix this indentation?

File: templates/supported/newscaster/butter/template.js [hunk 2]
```diff
@@ -6,7 +6,8 @@ document.addEventListener( "DOMContentLoaded", function( e ){
   Butter({
     config: "butter/config.json",
     ready: function( butter ){
-      var media = butter.media[ 0 ];
+      var media = butter.media[ 0 ],
+          popcorn = butter.media[ 0 ].popcorn.popcorn;
```
Comment (secretrobotron) [inline templates/supported/newscaster/butter/template.js]: This won't cause trouble, will it? I'm guessing you only need popcorn for this closure, and not when export happens?

File: templates/supported/newscaster/butter/template.js [hunk 3]
```diff
@@ -23,10 +24,114 @@ document.addEventListener( "DOMContentLoaded", function( e ){
...
+          butter.listen("trackeventadded", updateFunction);
+          butter.listen("trackeventupdated", updateFunction);
+
+        function updateFunction(e) {
+
+          var trackEvent,
+              _container = null,
+              _textEls,
+              _popcornOptions,
+              canvas = document.createElement( "canvas" ),
```
Comment (secretrobotron) [inline templates/supported/newscaster/butter/template.js]: Why the disparity between _ and no-_? Can you choose one and stick with it here?

File: templates/supported/newscaster/butter/template.js [hunk 4]
```diff
@@ -23,10 +24,114 @@ document.addEventListener( "DOMContentLoaded", function( e ){
...
+            trackEvent = e.target; 
+          } else { 
+            trackEvent = e; 
+          }
+
+          _popcornOptions = trackEvent.popcornOptions;
+
+          // Requires _container in options to work.
+          // This will be better when track events store a reference to popcorn in Butter, but it's ok for now.
+          trackEvent.popcornTrackEvent = popcorn.getTrackEvent( popcorn.getLastTrackEventId() ); //Store a reference
```
Comment (secretrobotron) [inline templates/supported/newscaster/butter/template.js]: Seriously? Please file a new bug for this if there isn't one already. 0.5.2. This is ridiculous.

File: templates/supported/newscaster/butter/template.js (hunk 2 above)
Comment (k88hudson) [inline templates/supported/newscaster/butter/template.js]: No, this is only inside butter. We will eventually have to deal with popcorn code, but for the purposes of this editing code, popcorn is always within the closure.

File: templates/supported/newscaster/butter/template.js (hunk 4 above)
Comment (k88hudson) [inline templates/supported/newscaster/butter/template.js]: There is a bug filed, but its in .6 i think

File: templates/supported/newscaster/butter/template.js [hunk 5]
```diff
@@ -1,33 +1,142 @@
...
+            }
+
+            if ( trackEvent.type === "image2" ) {
+              // Prevent default draggable behaviour of images
+              trackEvent.popcornTrackEvent._image.addEventListener( "mousedown", function(e) {
+                e.preventDefault();
+              }, false);
+
+              //Change default text to indicate draggable
+              if( !_popcornOptions.src && window.$ ){
```
Comment (humphd) [inline templates/supported/newscaster/butter/template.js:64]: you pass window.$ in as $ in the anonymous closure already.

File: templates/supported/newscaster/butter/template.js [hunk 6]
```diff
@@ -1,33 +1,142 @@
...
+                _container.innerHTML = "Drag an image from your desktop";
+              }
+
+              //Apply resizable/draggable if jQuery exists
+
+              if( $ ) {
```
Comment (humphd) [inline templates/supported/newscaster/butter/template.js:70]: This can fail if $ is really null.  Why would it be?  Don't you include jquery in the page?

**Pull request merged**